pkgs/by-name/te/textidote/package.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, ant
+, jdk11
+, makeWrapper
+, stripJavaArchivesHook
+, testers
+, textidote
+}:
+
+let
+  pname = "textidote";
+  version = "0.8.3";
+  jdk = jdk11;
+
+  # This is a fixed-output derivation; the Ant buildscript downloads dependencies and stores them alongside the source code:
+  # https://github.com/sylvainhalle/AntRun
+  # https://github.com/sylvainhalle/textidote/blob/c9abf447bec814d74954a08b8514b44cc972ac65/build.xml#L53-L64
+  populated-src = stdenv.mkDerivation {
+    pname = "textidote-populated-src";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "sylvainhalle";
+      repo = pname;
+      rev = "refs/tags/v${version}";
+      sha256 = "sha256-QMSoxk5jDn6qsdxffXJ/S4eTIzLjJdAEbdaK8MZIavI=";
+    };
+
+    nativeBuildInputs = [
+      ant
+      jdk
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      ant download-deps junit jacoco
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -a . $out
+
+      runHook postInstall
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHashAlgo = "sha256";
+    outputHash = "sha256-LrFClt8zN/ma42+Yoqwoy03TCuC3JfAeb02vehkljBo=";
+  }
+  ;
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = populated-src;
+
+  nativeBuildInputs = [
+    ant
+    jdk
+    makeWrapper
+    stripJavaArchivesHook
+  ];
+
+  # `javac` encoding defaults to UTF-8 after Java 18
+  # We can't use higher than Java 11 because that's the last version that supports targeting Java 6
+  env.ANT_OPTS = "-Dfile.encoding=utf8";
+
+  buildPhase = ''
+    runHook preBuild
+
+    ant jar
+
+    runHook postBuild
+  '';
+
+  doCheck = true;
+  checkPhase = ''
+    runHook preCheck
+
+    ant test
+
+    runHook postCheck
+  '';
+
+  # Recreates the wrapper from the `.deb`
+  # The `.deb` seems to be manually created for every release; there's no script in the repo
+  installPhase = ''
+    runHook preInstall
+
+    # copy generated jar file(s) to an appropriate location in $out
```

Comment:
```suggestion
```

1. Redundant comment
2. comments in Bash are a bit useless, since they change the code without really change a thing.